### ISSUE TITLE
Cromwell (large) workflow failed (NullPointerException among other reasons), but I must admit I am not sure why....

### ISSUE THREAD

**Issue by LeeTL1220** [broadinstitute/cromwell#2228]
- JES backend
- cromwell server
- localhost mysql
- cromwell-27-c89c83f-SNAP.jar
- I set the database queue size to 3000.
- I have *not* changed the metadata batch size.

*Should I attempt to restart this workflow?*  This took over 4 hours to get this error message and I do not want to incur the cost if it will fail the same way again.

Side issues:
- My workflow failed and yet cromwell is still *mauling* the mysql server.
- The call cache lookups are taking >1 hour per task.

Main issue:

I do not understand the error messages, but my workflow has entered a Failed state and I am not sure why.

First, I see a bunch of NPE:
```
[ERROR] [05/01/2017 17:36:00.055] [cromwell-system-akka.dispatchers.engine-dispatcher-84] [akka.dispatch.Dispatcher] null
java.lang.NullPointerException
        at cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheWriteActor.receiver(CallCacheWriteActor.scala:17)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheWriteActor$$anonfun$1.apply(CallCacheWriteActor.scala:21)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheWriteActor$$anonfun$1.apply(CallCacheWriteActor.scala:19)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:90)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:39)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:415)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

```

Then I see:

```
[ERROR] [05/01/2017 17:36:04.203] [cromwell-system-akka.dispatchers.engine-dispatcher-84] [akka://cromwell-system/user/cromwell-service/WorkflowManagerActor] WorkflowManagerActor Workflow 5
3e95ead-9026-4c13-89f9-f6c675214523 failed (during ExecutingWorkflowState): Task m2.Mutect2.M2:1:1 failed. JES error code 5.  Message: 9: Failed to localize files: failed to copy the follow
ing files: "gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam -> /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD
/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam (cp failed: gsutil -q -m cp gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A2
5E-08.2.bam /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam, command failed: Traceback (most recent call last):\n
 File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 75, in <module>\n    main()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.
py\", line 22, in main\n    project, account = bootstrapping.GetActiveProjectAndAccount()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py\", line 205,
in GetActiveProjectAndAccount\n    project_name = properties.VALUES.core.project.Get(validate=False)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py
\", line 1221, in Get\n    required)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1501, in _GetProperty\n    value = _GetPropertyWithoutD
efault(prop, properties_file)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1539, in _GetPropertyWithoutDefault\n    value = callback()\n
 File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 693, in _GetGCEProject\n    return c_gce.Metadata().Project()\n  File \"/usr/local/share/googl
e/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 104, in Project\n    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/g
ooglecloudsdk/core/util/retry.py\", line 155, in TryFunc\n    return func(*args, **kwargs), None\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.p
y\", line 41, in _ReadNoProxyWithCleanFailures\n    return gce_read.ReadNoProxy(uri)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py\", li
ne 50, in ReadNoProxy\n    request, timeout=timeout_property).read()\n  File \"/usr/lib/python2.7/urllib2.py\", line 401, in open\n    response = self._open(req, data)\n  File \"/usr/lib/py
thon2.7/urllib2.py\", line 419, in _open\n    '_open', req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 379, in _call_chain\n    result = func(*args)\n  File \"/usr/lib/python2.7/urllib
2.py\", line 1211, in http_open\n    return self.do_open(httplib.HTTPConnection, req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1184, in do_open\n    r = h.getresponse(buffering=True)
\n  File \"/usr/lib/python2.7/httplib.py\", line 1072, in getresponse\n    response.begin()\n  File \"/usr/lib/python2.7/httplib.py\", line 408, in begin\n    version, status, reason = self
._read_status()\n  File \"/usr/lib/python2.7/httplib.py\", line 366, in _read_status\n    line = self.fp.readline()\n  File \"/usr/lib/python2.7/socket.py\", line 447, in readline\n    data
 = self._sock.recv(self._rbufsize)\nsocket.timeout: timed out\n); gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam -> /mnt/loc
al-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam (cp failed: gsutil -q -m cp gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga
/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam,
 command failed: Traceback (most recent call last):\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 75, in <module>\n    main()\n  File \"/usr/local/s
hare/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 22, in main\n    project, account = bootstrapping.GetActiveProjectAndAccount()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py\", line 205, in GetActiveProjectAndAccount\n    project_name = properties.VALUES.core.project.Get(validate=False)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1221, in Get\n    required)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1501, in _GetProperty\n    value = _GetPropertyWithoutDefault(prop, properties_file)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1539, in _GetPropertyWithoutDefault\n    value = callback()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 693, in _GetGCEProject\n    return c_gce.Metadata().Project()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 104, in Project\n    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py\", line 155, in TryFunc\n    return func(*args, **kwargs), None\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 41, in _ReadNoProxyWithCleanFailures\n    return gce_read.ReadNoProxy(uri)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py\", line 50, in ReadNoProxy\n    request, timeout=timeout_property).read()\n  File \"/usr/lib/python2.7/urllib2.py\", line 401, in open\n    response = self._open(req, data)\n  File \"/usr/lib/python2.7/urllib2.py\", line 419, in _open\n    '_open', req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 379, in _call_chain\n    result = func(*args)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1211, in http_open\n    return self.do_open(httplib.HTTPConnection, req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1184, in do_open\n    r = h.getresponse(buffering=True)\n  File \"/usr/lib/python2.7/httplib.py\", line 1072, in getresponse\n    response.begin()\n  File \"/usr/lib/python2.7/httplib.py\", line 408, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python2.7/httplib.py\", line 366, in _read_status\n    line = self.fp.readline()\n  File \"/usr/lib/python2.7/socket.py\", line 447, in readline\n    data = self._sock.recv(self._rbufsize)\nsocket.timeout: timed out\n)"
java.lang.Exception: Task m2.Mutect2.M2:1:1 failed. JES error code 5.  Message: 9: Failed to localize files: failed to copy the following files: "gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam -> /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam (cp failed: gsutil -q -m cp gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-10A-01D-A25E-08.2.bam, command failed: Traceback (most recent call last):\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 75, in <module>\n    main()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 22, in main\n    project, account = bootstrapping.GetActiveProjectAndAccount()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py\", line 205, in GetActiveProjectAndAccount\n    project_name = properties.VALUES.core.project.Get(validate=False)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1221, in Get\n    required)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1501, in _GetProperty\n    value = _GetPropertyWithoutDefault(prop, properties_file)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1539, in _GetPropertyWithoutDefault\n    value = callback()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 693, in _GetGCEProject\n    return c_gce.Metadata().Project()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 104, in Project\n    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py\", line 155, in TryFunc\n    return func(*args, **kwargs), None\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 41, in _ReadNoProxyWithCleanFailures\n    return gce_read.ReadNoProxy(uri)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py\", line 50, in ReadNoProxy\n    request, timeout=timeout_property).read()\n  File \"/usr/lib/python2.7/urllib2.py\", line 401, in open\n    response = self._open(req, data)\n  File \"/usr/lib/python2.7/urllib2.py\", line 419, in _open\n    '_open', req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 379, in _call_chain\n    result = func(*args)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1211, in http_open\n    return self.do_open(httplib.HTTPConnection, req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1184, in do_open\n    r = h.getresponse(buffering=True)\n  File \"/usr/lib/python2.7/httplib.py\", line 1072, in getresponse\n    response.begin()\n  File \"/usr/lib/python2.7/httplib.py\", line 408, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python2.7/httplib.py\", line 366, in _read_status\n    line = self.fp.readline()\n  File \"/usr/lib/python2.7/socket.py\", line 447, in readline\n    data = self._sock.recv(self._rbufsize)\nsocket.timeout: timed out\n); gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam -> /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam (cp failed: gsutil -q -m cp gs://5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam /mnt/local-disk/5aa919de-0aa0-43ec-9ec3-288481102b6d/tcga/STAD/DNA/WXS/BI/ILLUMINA/C440.TCGA-HU-A4H3-01A-21D-A25D-08.1.bam, command failed: Traceback (most recent call last):\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 75, in <module>\n    main()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 22, in main\n    project, account = bootstrapping.GetActiveProjectAndAccount()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py\", line 205, in GetActiveProjectAndAccount\n    project_name = properties.VALUES.core.project.Get(validate=False)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1221, in Get\n    required)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1501, in _GetProperty\n    value = _GetPropertyWithoutDefault(prop, properties_file)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1539, in _GetPropertyWithoutDefault\n    value = callback()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 693, in _GetGCEProject\n    return c_gce.Metadata().Project()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 104, in Project\n    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py\", line 155, in TryFunc\n    return func(*args, **kwargs), None\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 41, in _ReadNoProxyWithCleanFailures\n    return gce_read.ReadNoProxy(uri)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py\", line 50, in ReadNoProxy\n    request, timeout=timeout_property).read()\n  File \"/usr/lib/python2.7/urllib2.py\", line 401, in open\n    response = self._open(req, data)\n  File \"/usr/lib/python2.7/urllib2.py\", line 419, in _open\n    '_open', req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 379, in _call_chain\n    result = func(*args)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1211, in http_open\n    return self.do_open(httplib.HTTPConnection, req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1184, in do_open\n    r = h.getresponse(buffering=True)\n  File \"/usr/lib/python2.7/httplib.py\", line 1072, in getresponse\n    response.begin()\n  File \"/usr/lib/python2.7/httplib.py\", line 408, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python2.7/httplib.py\", line 366, in _read_status\n    line = self.fp.readline()\n  File \"/usr/lib/python2.7/socket.py\", line 447, in readline\n    data = self._sock.recv(self._rbufsize)\nsocket.timeout: timed out\n)"
        at cromwell.backend.impl.jes.JesAsyncBackendJobExecutionActor$.StandardException(JesAsyncBackendJobExecutionActor.scala:63)
        at cromwell.backend.impl.jes.JesAsyncBackendJobExecutionActor.handleExecutionFailure(JesAsyncBackendJobExecutionActor.scala:411)
        at cromwell.backend.impl.jes.JesAsyncBackendJobExecutionActor.handleExecutionFailure(JesAsyncBackendJobExecutionActor.scala:67)
        at cromwell.backend.standard.StandardAsyncExecutionActor$$anonfun$handleExecutionResult$1.applyOrElse(StandardAsyncExecutionActor.scala:666)
        at cromwell.backend.standard.StandardAsyncExecutionActor$$anonfun$handleExecutionResult$1.applyOrElse(StandardAsyncExecutionActor.scala:663)
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:346)
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:345)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:90)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:39)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:415)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

```

**Comment by LeeTL1220** [comment]
And again on a new run, with something that looks very similar...

```
[ERROR] [05/01/2017 21:06:38.897] [cromwell-system-akka.dispatchers.engine-dispatcher-86] [akka.dispatch.Dispatcher] null
java.lang.NullPointerException
        at cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheWriteActor.receiver(CallCacheWriteActor.scala:17)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheWriteActor$$anonfun$1.apply(CallCacheWriteActor.scala:21)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheWriteActor$$anonfun$1.apply(CallCacheWriteActor.scala:19)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:90)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:39)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:415)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

[ERROR] [05/01/2017 21:06:41.921] [cromwell-system-akka.dispatchers.engine-dispatcher-106] [akka://cromwell-system/user/cromwell-service/WorkflowManagerActor] WorkflowManagerActor Workflow
67fdb82c-72bb-4d33-a74b-441a8db2a780 failed (during ExecutingWorkflowState): Task m2.Mutect2.M2:108:1 failed. JES error code 10.  Message: 15: Gsutil failed: failed to upload logs for "gs:/
/broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full_dl_ob_training_with_m2/67fdb82c-72bb-4d33-a74b-441a8db2a780/call-m2_nt/shard-37/Mutect2/71720e5e-1769-46e7-a2b8-98d19
ec38f93/call-M2/shard-108/": cp failed: gsutil -h Content-type:text/plain -q -m cp /var/log/google-genomics/*.log gs://broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full
_dl_ob_training_with_m2/67fdb82c-72bb-4d33-a74b-441a8db2a780/call-m2_nt/shard-37/Mutect2/71720e5e-1769-46e7-a2b8-98d19ec38f93/call-M2/shard-108/, command failed: Traceback (most recent call
 last):
  File "/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py", line 75, in <module>
    main()
  File "/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py", line 22, in main
    project, account = bootstrapping.GetActiveProjectAndAccount()
  File "/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py", line 205, in GetActiveProjectAndAccount
    project_name = properties.VALUES.core.project.Get(validate=False)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1221, in Get
    required)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1501, in _GetProperty
    value = _GetPropertyWithoutDefault(prop, properties_file)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1539, in _GetPropertyWithoutDefault
    value = callback()
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 693, in _GetGCEProject
    return c_gce.Metadata().Project()
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py", line 104, in Project
    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py", line 155, in TryFunc
    return func(*args, **kwargs), None
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py", line 41, in _ReadNoProxyWithCleanFailures
    return gce_read.ReadNoProxy(uri)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py", line 50, in ReadNoProxy
    request, timeout=timeout_property).read()
  File "/usr/lib/python2.7/urllib2.py", line 401, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 419, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 379, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 1211, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.7/urllib2.py", line 1184, in do_open
    r = h.getresponse(buffering=True)
  File "/usr/lib/python2.7/httplib.py", line 1072, in getresponse
    response.begin()
  File "/usr/lib/python2.7/httplib.py", line 408, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.7/httplib.py", line 366, in _read_status
    line = self.fp.readline()
  File "/usr/lib/python2.7/socket.py", line 447, in readline
    data = self._sock.recv(self._rbufsize)
socket.timeout: timed out
:
```

**Comment by LeeTL1220** [comment]
From the metadata:
```
"failures": [
    {
      "causedBy": [],
      "message": "Task m2.Mutect2.M2:108:1 failed. JES error code 10.  Message: 15: Gsutil failed: failed to upload logs for \"gs://broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full_dl_ob_training_with_m2/67fdb82c-72bb-4d33-a74b-441a8db2a780/call-m2_nt/shard-37/Mutect2/71720e5e-1769-46e7-a2b8-98d19ec38f93/call-M2/shard-108/\": cp failed: gsutil -h Content-type:text/plain -q -m cp /var/log/google-genomics/*.log gs://broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full_dl_ob_training_with_m2/67fdb82c-72bb-4d33-a74b-441a8db2a780/call-m2_nt/shard-37/Mutect2/71720e5e-1769-46e7-a2b8-98d19ec38f93/call-M2/shard-108/, command failed: Traceback (most recent call last):\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 75, in <module>\n    main()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py\", line 22, in main\n    project, account = bootstrapping.GetActiveProjectAndAccount()\n  File \"/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py\", line 205, in GetActiveProjectAndAccount\n    project_name = properties.VALUES.core.project.Get(validate=False)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1221, in Get\n    required)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1501, in _GetProperty\n    value = _GetPropertyWithoutDefault(prop, properties_file)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 1539, in _GetPropertyWithoutDefault\n    value = callback()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py\", line 693, in _GetGCEProject\n    return c_gce.Metadata().Project()\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 104, in Project\n    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py\", line 155, in TryFunc\n    return func(*args, **kwargs), None\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py\", line 41, in _ReadNoProxyWithCleanFailures\n    return gce_read.ReadNoProxy(uri)\n  File \"/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py\", line 50, in ReadNoProxy\n    request, timeout=timeout_property).read()\n  File \"/usr/lib/python2.7/urllib2.py\", line 401, in open\n    response = self._open(req, data)\n  File \"/usr/lib/python2.7/urllib2.py\", line 419, in _open\n    '_open', req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 379, in _call_chain\n    result = func(*args)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1211, in http_open\n    return self.do_open(httplib.HTTPConnection, req)\n  File \"/usr/lib/python2.7/urllib2.py\", line 1184, in do_open\n    r = h.getresponse(buffering=True)\n  File \"/usr/lib/python2.7/httplib.py\", line 1072, in getresponse\n    response.begin()\n  File \"/usr/lib/python2.7/httplib.py\", line 408, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python2.7/httplib.py\", line 366, in _read_status\n    line = self.fp.readline()\n  File \"/usr/lib/python2.7/socket.py\", line 447, in readline\n    data = self._sock.recv(self._rbufsize)\nsocket.timeout: timed out\n: "
    }
```

**Comment by Horneth** [comment]
So the real reason of the failure I'm afraid is the JES one, which I don't think we can do much about except tell them about it ?
The Cromwell NPE is still a bug but I believe is an artifact of the job failing combined with the fact that the DB is swamped.

**Comment by LeeTL1220** [comment]
@Horneth I pruned my number of jobs dramatically....  I am now down to <5k.  And I am still getting this error and very slow call caching.  @katevoss Can we tell google, if the crashes are on their end?

**Comment by katevoss** [comment]
@LeeTL1220 can you summarize what the Google crash(es) are about so I can correctly communicate it to them? They'll be in the office on Friday conveniently.

**Comment by LeeTL1220** [comment]
@katevoss I think you'd prefer @Horneth to do that.  I'm not sure what the problem is.

**Comment by LeeTL1220** [comment]
Hey @katevoss and @Horneth ... I dropped my call caching database to help with speed and reran.  Now I was doing < 500 jobs at a time.  I still got that JES error.

```
"failures": [
    {
      "causedBy": [],
      "message": "Task dl_ob_training.dl_ob_training.VcfToIntervals:NA:1 failed. JES error code 5.  Message: 10: Failed to delocalize files: failed to copy the following files: \"/mnt/local-disk/C499.TCGA-EL-A3GX-01A-11D-A202-08.1_variant_intervals.list -> gs://broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full_dl_ob_training_with_m2/a6267914-4975-479c-ad64-6e18f9735ee8/call-dl_ob_training_mc3/shard-0/dl_ob_training/1cebfb98-3272-47b8-ae2f-e13e518ca2b7/call-VcfToIntervals/C499.TCGA-EL-A3GX-01A-11D-A202-08.1_variant_intervals.list (cp failed: gsutil -q -m cp -L /var/log/google-genomics/out.log /mnt/local-disk/C499.TCGA-EL-A3GX-01A-11D-A202-08.1_variant_intervals.list gs://broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full_dl_ob_training_with_m2/a6267914-4975-479c-ad64-6e18f9735ee8/call-dl_ob_training_mc3/shard-0/dl_ob_training/1cebfb98-3272-47b8-ae2f-e13e518ca2b7/call-VcfToIntervals/C499.TCGA-EL-A3GX-01A-11D-A202-08.1_variant_intervals.list, command failed: CommandException: No URLs matched: /mnt/local-disk/C499.TCGA-EL-A3GX-01A-11D-A202-08.1_variant_intervals.list\\nCommandException: 1 file/object could not be transferred.\\n)\""
    }
  ],

```

**Comment by Horneth** [comment]
```
[ERROR] [05/01/2017 21:06:41.921] [cromwell-system-akka.dispatchers.engine-dispatcher-106] [akka://cromwell-system/user/cromwell-service/WorkflowManagerActor] WorkflowManagerActor Workflow
67fdb82c-72bb-4d33-a74b-441a8db2a780 failed (during ExecutingWorkflowState): Task m2.Mutect2.M2:108:1 failed. JES error code 10.  Message: 15: Gsutil failed: failed to upload logs for "gs:/
/broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full_dl_ob_training_with_m2/67fdb82c-72bb-4d33-a74b-441a8db2a780/call-m2_nt/shard-37/Mutect2/71720e5e-1769-46e7-a2b8-98d19
ec38f93/call-M2/shard-108/": cp failed: gsutil -h Content-type:text/plain -q -m cp /var/log/google-genomics/*.log gs://broad-dsde-methods/lichtens/cromwell-executions-test-dl-oxoq-full/full
_dl_ob_training_with_m2/67fdb82c-72bb-4d33-a74b-441a8db2a780/call-m2_nt/shard-37/Mutect2/71720e5e-1769-46e7-a2b8-98d19ec38f93/call-M2/shard-108/, command failed: Traceback (most recent call
 last):
  File "/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py", line 75, in <module>
    main()
  File "/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/gsutil.py", line 22, in main
    project, account = bootstrapping.GetActiveProjectAndAccount()
  File "/usr/local/share/google/google-cloud-sdk/bin/bootstrapping/bootstrapping.py", line 205, in GetActiveProjectAndAccount
    project_name = properties.VALUES.core.project.Get(validate=False)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1221, in Get
    required)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1501, in _GetProperty
    value = _GetPropertyWithoutDefault(prop, properties_file)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 1539, in _GetPropertyWithoutDefault
    value = callback()
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/properties.py", line 693, in _GetGCEProject
    return c_gce.Metadata().Project()
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py", line 104, in Project
    gce_read.GOOGLE_GCE_METADATA_PROJECT_URI)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/util/retry.py", line 155, in TryFunc
    return func(*args, **kwargs), None
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce.py", line 41, in _ReadNoProxyWithCleanFailures
    return gce_read.ReadNoProxy(uri)
  File "/usr/local/share/google/google-cloud-sdk/lib/googlecloudsdk/core/credentials/gce_read.py", line 50, in ReadNoProxy
    request, timeout=timeout_property).read()
  File "/usr/lib/python2.7/urllib2.py", line 401, in open
    response = self._open(req, data)
  File "/usr/lib/python2.7/urllib2.py", line 419, in _open
    '_open', req)
  File "/usr/lib/python2.7/urllib2.py", line 379, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 1211, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/lib/python2.7/urllib2.py", line 1184, in do_open
    r = h.getresponse(buffering=True)
  File "/usr/lib/python2.7/httplib.py", line 1072, in getresponse
    response.begin()
  File "/usr/lib/python2.7/httplib.py", line 408, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.7/httplib.py", line 366, in _read_status
    line = self.fp.readline()
  File "/usr/lib/python2.7/socket.py", line 447, in readline
    data = self._sock.recv(self._rbufsize)
socket.timeout: timed out
:
```

This is a gsutil stacktrace. JES tried to copy the logs and failed, hence failing the job and the workflow. They might want to retry this - although we've been telling them to stop retrying too much on some things so I don't know. @geoffjentry and @cjllanwarne  were talking about it on slack maybe they have an opinion.

For call caching: it will get slower and slower. Basically the more jobs you run the slower it's going to be... I'm working on something to fix that but it's not in develop yet.

**Comment by LeeTL1220** [comment]
I dropped my database and call caching sped up.

**Comment by LeeTL1220** [comment]
But I'm having trouble running even 500 tasks without one of these JES failures.  Is the fact that I using  pre-emptible instances matter?

**Comment by Horneth** [comment]
Hmm I don't know, I don't think so. Those failures started popping up recently I believe

**Comment by Horneth** [comment]
@LeeTL1220 I'm seeing the same log copying failure in our test suite actually. So it very likely isn't your doing.

**Comment by patmagee** [comment]
@LeeTL1220 @Horneth I ran into similar issues when running > 400 Tasks Simultaneously. I would occasionally get 403 from the JES backend from various causes
- size built in function would timeout  
- Pulling the docker image from gcr.io would time out
- occasionally pulling the docker image from docker hub would also time out
- I also observed the above error that you were experiencing as well

I was not able to debug any of them, because of the transient nature. Rerunning the workflow generally fixed the problem. I am also using preemptible instances for the majority of the tasks that were being run, however I do not see how that could be contributing to the issue. If anything i would guess that we are bumping into an api quota and are being throttled by google leading to the timeouts

**Comment by LeeTL1220** [comment]
@Horneth @patmagee I'm going to have to enable continue on failures, except I am not sure that is valid.
@Horneth Can I tell cromwell to continue even on these failures?  These are not non-zero return codes, right?
 
I actually can't just rerun the jobs and expect any progress.  Due to the high rate of these failures and that I scatter over quite a few jobs, nothing gets to complete before one job fails.  If I were to make my scatter narrower, I'd probably still get failures due, since each job would take substantially longer.  

In other words, I can't proceed unless there is a viable "continue on failure" option (or a failure in my logic).  And even then, I am likely to proceed slowly.

**Comment by patmagee** [comment]
@LeeTL1220 @Horneth I doubt enabling continue on return would work. You are getting timeouts not only when uploading log files, but also when localizing files. Ive observed this occasionally to with wide scatters and multiple workflows. 
It starting to seem more like an api Issue. I know in the cromwell conf there is a property for setting the total number of concurrent workflows, but I do not know if this is extended to the task level. It would be interesting to see whether or not limiting the number of concurrent tasks in a scatter would have any impact on this. That or better scattering the task submission for scatters instead of submitting all tasks basically at once.

This is one of our major pain points too. So far the only reasonable solution we have had (other then adjusting api quotas) is just to tell users to rerun a wf

**Comment by geoffjentry** [comment]
@LeeTL1220 @patmagee I'd been paying less attention to this than I should have. I mentioned to @LeeTL1220 yesterday that the general issue of "sometimes logs fail to upload and it doesn't retry a ton - but Google knows this". However looking at the frequency and the actual errors makes me think I need to ping Google to make sure they know about the errors themselves. Perhaps there's something more fundamentally wrong going on that they're blissfully unaware of.

**Comment by geoffjentry** [comment]
I've filed a ticket w/ Google. Will report back when I get further info.

**Comment by Horneth** [comment]
@LeeTL1220 @patmagee you can use the `workflow_failure_mode` workflow option and set it to `ContinueWhilePossible`. This will keep the workflow running as long as it can: some jobs are running or their upstream hasn't failed (yet).
This should at least let all your shards finish even if one fails.

**Comment by geoffjentry** [comment]
@LeeTL1220 @patmagee The google folks need JES operation IDs. If this happens again please send some to me.

**Comment by LeeTL1220** [comment]
@geoffjentry Another issue (unrelated to cromwell) came up that blocks me from running these workflows.  I'll get the Operations ID as soon as this other problem is sorted.

**Comment by ruchim** [comment]
@LeeTL1220 It seems this issue has devolved quite a bit from the original NPE and since its been over a year, I'm closing this issue. There's a lot of different concerns in this issue so please re-open a fresh issue with any recent concerns. Thanks!
